### PR TITLE
Split out the hostname when default dhcp_domain is used in nova.conf

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

When /etc/nova/nova.conf does not have specify dhcp_domain to empty
string, a default string of '.novalocal' is returned by the meta data
service. So we need to just split the string and pick the first one
in the array.

```
$ curl http://169.254.169.254/latest/meta-data/hostname
testvm-1.novalocal
$ curl http://169.254.169.254/latest/meta-data/public-hostname
testvm-1.novalocal
$ curl http://169.254.169.254/latest/meta-data/local-hostname
testvm-1.novalocal
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FengyunPan2** [comment]
Nice catch.
/lgtm

**Comment by dims** [comment]
/test pull-kubernetes-verify

hmm, logs are lost.

**Review by anguslees — COMMENTED** [review]
Oh no, when did we confuse the node name and hostname again?

**Comment by dims** [comment]
@anguslees pretty harmless change right .. let's please get this in?

**Comment by voelzmo** [comment]
The original bug why the switch from vm name to `hostname` was done was this: https://github.com/kubernetes/kubernetes/issues/57765

This bug was raised because vm names can include arbitrary symbols, including slashes, spaces, etc, which aren't possible for DNS names. Therefore, your fix now only works, if `hostname` minus the local domain is equal to vm name – which is only true if the vm name doesn't contain any of those 'bad symbols' in the first place.

TL;DR: we're now back to square one, where vm names may not contain any symbols which aren't allowed in DNS names.

What about my first suggestion in https://github.com/kubernetes/kubernetes/issues/57765#issuecomment-356526418 to use the UUID instead?

**Comment by dims** [comment]
https://github.com/kubernetes/kubernetes/issues/62223
